Examples/Examples-iOS/IGListKitExamples/ViewControllers/MixedDataViewController.swift

```diff
@@ -72,18 +75,27 @@ final class MixedDataViewController: UIViewController, ListAdapterDataSource {
     // MARK: ListAdapterDataSource
 
     func objects(for listAdapter: ListAdapter) -> [ListDiffable] {
-        guard selectedClass != nil else {
-            return data.map { $0 as! ListDiffable }
+        guard let selectedClass = selectedClass else {
+            return data.map {
+                return $0
+            }
         }
-        return data.filter { type(of: $0) == selectedClass! }
-            .map { $0 as! ListDiffable }
+        return data.filter {
+                if selectedClass == String.self, $0 is String {
+                    return true
+                }
+                return type(of: $0) == selectedClass
```

Comment:
Isn't this condition a superset of the one above?  Also -- not a big fan of the double indent above.